### ISSUE TITLE
staggered_dslash_test fails for daggered

### ISSUE THREAD

**Issue by mathiaswagner** [lattice/quda#267]
As I am exploring the parameter space for automated testing I noticed:

```
 ./staggered_dslash_test --dagger
[...]
[ RUN      ] dslash.verify
0 fails = 165888
1 fails = 165888
2 fails = 165888
3 fails = 165888
4 fails = 165888
5 fails = 165888
1.000000e-01 Failures: 995261 / 995328  = 9.999327e-01
1.000000e-02 Failures: 995322 / 995328  = 9.999940e-01
1.000000e-03 Failures: 995328 / 995328  = 1.000000e+00
[...]
1.000000e-16 Failures: 995328 / 995328  = 1.000000e+00
```

I am not sure whether this is a known (and undocumented) issue and it is probably an issue with the tests / reference.

**Comment by mikeaclark** [comment]
I suspect this is an easy fix somewhere in the interface to either the GPU interface code or the CPU test code.  Some suggestions:
* does this problem happen for both multi-gpu and serial builds?
* does it happen for naive staggered (`--dslash_type staggered`)?
* does it happen for `--test 1`? 
* can you ascertain whether the error ss coming from the CPU or GPU code?  E.g., compare with non-daggered and see if either the CPU or GPU results match the daggered result, which would indicate a lack of daggering?

Marking as a 0.7.1 bug.

**Comment by mathiaswagner** [comment]
Forcing `QUDA_DAG_NO` for the FPU reference *fixes* the issue, i.e.:
```
./staggered_dslash_test --dagger 
```
and
```
./staggered_dslash_test 
```

pass the test. 

`--test 1` does not matter.

I have not yet tried multi-GPU.

**Comment by mikeaclark** [comment]
Ok, I've worked out the problem.  The GPU code doesn't actually build a dagger kernel, though the macro option is in `staggered_dslash_core.h`, it's never defined.  For a CG solver, it uses the fact that M^dagger M = m^2 + D^dagger D = m^2 - D^2, so the dslash dagger kernel isn't needed.  There's no need to add the dagger kernel, and all it will do is increase compile time unnecessarily.  So I think we just need to document the fact that the dagger option doesn't apply to the staggered dslash.

**Comment by mathiaswagner** [comment]
At some point I was wondering about the need for the dagger for staggered but was not sure whether I missed some possible use. 
The existence of the dagger for the test code made me assume that this might be the case.
Probably best to throw an error if someone wants to use dagger for staggered. Unless the (probably runtime check) will cause a performance impact.

**Comment by mathiaswagner** [comment]
I removed it from 0.7.1 as nothing is really broken in QUDA and it should not stop us from releasing 0.7.1.

**Comment by maddyscientist** [comment]
Given the increased testing and dslash rewrite, this bug will be closed with #776.
